chore: update gitlab dependencies to reflect latest webhook payload

xNok opened the pull request:
🧽 Update or clean up current behavior

The issue mostly lies in using the `go-playground/webhooks` module, as the event parsing is never up-to-date. In fact, I just realised that I last contributed to it, but I no longer intend to do so because it is much better to use the Gitlab-maintained go-gitlab module.

The new approach should make the event Source compatible with all new event types (it was missing quite a few), and update the go-gitlab modules periodically is all you need to do to support the latest Gitlab development.

I didn't apply it, but it would be nice if the events contained an indication of what version of Gitlab or the go-gitlab module it is using as source, as only getting the benefit if they are up-to-date.

## Proposed Changes

- Upgrade go-gitlab (it was relocated to Gitlab)
- Update the Adapter logic to use the go-gitlab module

**Release Note**

<!--
🗒️ If this change has user-visible impact, write a release note in the block
below. Include the string "action required" if additional action is required of
users switching to the new release, for example in case of a breaking change.

Write as if you are speaking to users, not other Knative contributors. If this
change has no user-visible impact, no release-note is needed.
-->

```release-note
GitlabSource now support all Gitlab Webhook Event Types to their latest version. 

* [EventTypes](https://gitlab.com/gitlab-org/api/client-go/-/blob/main/types.go?ref_type=heads#L693)
* [EventPayloads](https://gitlab.com/gitlab-org/api/client-go/-/blob/main/event_webhook_types.go?ref_type=heads)

You need to update your event Sink dependency to see the results, which might include migrating from `go-playground/webhooks` to `gitlab.com/gitlab-org/api/client-go`. As `go-playground/webhooks` is not keeping up with Gitlab updates.
```

**Docs**

<!--
📖 If this change has user-visible impact, link to an issue or PR in
https://github.com/knative/docs.
-->

PabloPie commented:
Hi @aslom @pierDipi! Do you think you could have a look at this MR? Thank you!

Cali0707 reviewed (commented):
Thanks for working on updating this @xNok !

Would you be able to fix the linter errors re: the license boilerplate in the new `webhook.go` file?

Overall, this PR looks good to me otherwise. Have you managed to test it in a k8s cluster and see if events show up from the webhook?

xNok commented:
Thanks for having a look, I will get back to it on Friday for further testing and wrap this up

xNok commented:
@Cali0707 I have updated the license in the `webhook.go` and, more importantly, implemented e2e tests for the webhook reciever part.

I am running the e2e test against the knative quickstart kind cluster, I didn't hook it to the `[presubmit-tests.sh](https://github.com/knative-extensions/eventing-gitlab/blob/main/test/presubmit-tests.sh)

Cali0707 reviewed (commented):
Overall this looks good to me, with a few small nits

/lgtm
/cc @creydr

xNok commented on pkg/adapter/webhook.go:
```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2020 The Knative Authors.
```
```suggestion
Copyright 2025 The Knative Authors.
```

xNok commented on pkg/adapter/webhook.go:
```diff
@@ -0,0 +1,117 @@
+	EventSender EventSender
+}
+
+// webhookExample shows how to create a Webhook server to parse Gitlab events.
```
```suggestion
// NewWebhookHandler provide a webhook receiver that parses Gitlab events and emits CloudEvents.
```

dprotaso commented:
I rebased the PR
